tests/unit/mock_hostmod.c

```diff
@@ -233,9 +269,16 @@ osd_result osd_hostmod_reg_write(struct osd_hostmod_ctx *ctx,
         ck_assert_int_eq(exp->flags, flags);
     }
 
-    assert(reg_size_bit == 16); // XXX: Extend for other register sizes
-    uint64_t written_reg_val = *(uint16_t*)reg_val;
-    ck_assert_uint_eq(written_reg_val, exp->reg_val);
+    assert(reg_size_bit%16 == 0 && reg_size_bit<=32); // XXX: Extend for other register sizes (64 and 128 bits)
+    
+    if (reg_size_bit == 16) {
+        uint64_t written_reg_val = *(uint16_t*)reg_val;
+        ck_assert_uint_eq(written_reg_val, exp->reg_val);
+    } else {
+        uint64_t written_reg_val = *(uint32_t*)reg_val;
+        ck_assert_uint_eq(written_reg_val, exp->reg_val);
+    }
```

Comment:
Why not go ahead and extend this for 64 and 128 bit as well while you're at it?

Reply:
I wanted to extend it to 128 bits, but then I might need to change the size of "written_reg_val" and reg_val in all the programs. Also, or1k needs only 32 bit wide registers. So, I thought of skipping that part. But, I will go ahead and try to add those changes to ensure the CDM module support registers with width more than 32 bits,

Reply:
Please give it a try; if you figure out it requires an unfeasibly large amount of changes let me know (ideally with a work in progress diff). 64 bit shouldn't be a problem though (and is more important anyways).

Reply:
The program is working successfully for 64 bit wide registers. To add read/write tests for 128 bits wide registers, I think we need to add some external resource. 
Something like this: https://www.codeproject.com/Tips/784635/UInt-Bit-Operations
Please correct me if I am wrong. But, since there aren't many CPU cores with 128 bit wide registers, I think it is safe to move ahead.

Reply:
OK, then please add tests for 16, 32 and 64 bit architectures to get full coverage on the supported ones, and specify clearly that 128 bit register accesses are not supported right now. Use assertations and documentation to make this limitation obvious to the user.